### PR TITLE
Add support for default config override

### PR DESCRIPTION
**Pull request by dannyvv**
## Overview
This change adds another top level option to `just-scripts`. It already had a `--config` option to explicitly specify a config file. That setting overrides the config file to use and if not specified, each project must specify a config file next to package.json. This changes adds a new config option `--default-config` that allows a config file to be specified as last fallback. i.e. if a project has a local just config file, that one will be picked up. If not, it will fall back to the one specified in  `--default-config`. 

This is very useful for large monorepo's where projects are very homogeneous and all just.config.ts files look alike, yet still allows for extension and customization should a project need it.

## Example
This allows teams to add a package with a bin-script file in the shared repo i.e. `my-scripts.js` that can be used as `my-scripts build` instead of `just-scripts build`. The contents of my-scripts.js could be something like:
```
import * as path from 'path';
process.argv.push("--default-config", path.resolve(__dirname, "my.default.config.js"));
require("just-scripts/bin/just-scripts.js");
```
and have `my.default.config.js` as the default settings for the repo.

And projects that want to slightly customize can import the `my.default.config.js` and then override a few tasks, or fully specify the tasks they want.

## Test Notes
Factored the logic that determines which config file should be picked up and added unittest for those.

### CODE REVIEW

File: packages/just-task/src/__tests__/resolve.spec.ts [hunk 1]
```diff
@@ -186,3 +187,97 @@ describe('resolve', () => {
     expect(resolve('b.txt', 'd')).toContain(path.join('d', 'b.txt'));
   });
 });
+
+
+describe('resolveCwd', () => {
```
Comment (dannyvv) [inline packages/just-task/src/__tests__/resolve.spec.ts:192]: Will remove this, accidentally copy pasted twice...

Comment (dannyvv) [reply packages/just-task/src/__tests__/resolve.spec.ts]: Fixed

File: packages/just-task/src/__tests__/resolve.spec.ts [hunk 2]
```diff
@@ -186,3 +187,58 @@ describe('resolve', () => {
...
+
+  it('default chooses local config', () => {
+    mockfs({
+      config: {
+        'configArgument.ts': 'formConfig',
+        'defaultConfigArgument.ts': 'formDefaultConfig',
+      },
+      'just.config.ts': 'localConfig',
+    });
+    const resolvedConfig = config.resolveConfigFile({ config: undefined, 'default-config': undefined } as any)
```
Comment (ecraig12345) [inline packages/just-task/src/__tests__/resolve.spec.ts:205]: Should `'default-config'` here be `defaultConfig`?

Comment (dannyvv) [reply packages/just-task/src/__tests__/resolve.spec.ts:205]: @ecraig12345: Thanks! yes it should have been.

**Review by ecraig12345 — APPROVED** [review]

**Pull request merged**